ubuntu/16.04/usr/local/share/bootstrap/update_permissions.sh

```diff
@@ -17,12 +17,21 @@ function update_permissions() {
     REFERENCE="$1"
     read -r owner group owner_id group_id < <(stat -c '%U %G %u %g' "$REFERENCE")
     if [[ "$owner" = UNKNOWN ]]; then
-        APP_USER="$(randname)"
+        APP_USER="build"
+        if [ "$APP_USER_LOCAL_RANDOM" == 'true' ]; then
+            APP_USER="$(randname)"
+        fi
+        CODE_OWNER="$APP_USER"
         export APP_USER
+        export CODE_OWNER
+
         if [[ "$group" = UNKNOWN ]]; then
-            export APP_GROUP="$owner"
-            addgroup --system --gid "$group_id" "$APP_GROUP"
+            APP_GROUP="$owner"
+            CODE_GROUP="$APP_GROUP"
+            export APP_GROUP
+            export CODE_GROUP
+            groupadd --system --gid "$group_id" "$APP_GROUP" || groupmod -g "$group_id" "$APP_GROUP"
         fi
-        adduser --system --uid="$owner_id" --gid="$group_id" "$APP_USER"
+        useradd --create-home --system --uid "$owner_id" --gid "$group_id" "$APP_USER" || usermod -u "$owner_id" "$APP_USER"
```

Comment:
A very good point that I hadn't considered.

http://manpages.ubuntu.com/manpages/xenial/man8/usermod.8.html says the following:
```
           The user's mailbox, and any files which the user owns and which are
           located in the user's home directory will have the file user ID
           changed automatically.

           The ownership of files outside of the user's home directory must be
           fixed manually.
```

It might be safe to say that everything outside the home directory will be in /app in the case of our image (unless APP_USER is set to a system user).